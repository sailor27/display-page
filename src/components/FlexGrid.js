import React from 'react';
import PropTypes from 'prop-types';
import Product from './Product';
import {makeStyles} from '@material-ui/core';

const useStyles = makeStyles({
    root: {
        marginTop: 50,
        display: 'flex',
        flexFlow: 'row wrap',
        width: '100vw',
        justifyContent: 'center',
    }
});

function FlexGrid({products}) {
    const classes = useStyles();

    return (
        <div className={classes.root}>
            {products.map(product => (
                <Product
                    key={product.id}
                    product={product}
                />
            ))}
        </div>
    );
}

FlexGrid.propTypes = {
    products: PropTypes.array,
};

export default FlexGrid;

