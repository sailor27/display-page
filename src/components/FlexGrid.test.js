import React from 'react';
import ReactDOM from 'react-dom';
import {render, cleanup} from '@testing-library/react';
import * as PRODUCTS from '../data/shop-all-new.json';
import FlexGrid from './FlexGrid';

describe('FlexGrid', () => {
    afterEach(() => cleanup());

    test('renders without crashing', () => {
        const div = document.createElement('div');

        ReactDOM.render(<FlexGrid products={PRODUCTS.groups} />, div);
    });

    test('matches snapshot', () => {
        expect.assertions(1);

        const {asFragment} = render(<FlexGrid products={PRODUCTS.groups} />);

        expect(asFragment()).toMatchSnapshot();
    });
});
