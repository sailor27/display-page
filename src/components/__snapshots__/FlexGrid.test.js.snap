// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlexGrid matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="MuiPaper-root MuiCard-root makeStyles-card-2 makeStyles-card-104 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <button
        class="MuiButtonBase-root MuiCardActionArea-root makeStyles-cardAction-3"
        tabindex="0"
        type="button"
      >
        <div
          class="MuiCardMedia-root makeStyles-media-4 makeStyles-media-105"
          style="background-image: url(https://www.westelm.com/weimgs/ab/images/wcm/products/202005/0001/urban-summer-outdoor-2-piece-chaise-sectional-m.jpg);"
          title="Urban Outdoor 2-Piece Chaise Sectional"
        />
        <h6
          class="MuiTypography-root makeStyles-label-5 MuiTypography-h6 MuiTypography-alignLeft"
        >
          Urban Outdoor 2-Piece Chaise Sectional
        </h6>
        <span
          class="MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="MuiPaper-root MuiCard-root makeStyles-card-2 makeStyles-card-106 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <button
        class="MuiButtonBase-root MuiCardActionArea-root makeStyles-cardAction-3"
        tabindex="0"
        type="button"
      >
        <div
          class="MuiCardMedia-root makeStyles-media-4 makeStyles-media-107"
          style="background-image: url(https://www.westelm.com/weimgs/ab/images/wcm/products/202005/0002/urban-summer-outdoor-sofa-81-1-m.jpg);"
          title="Urban Outdoor Sofa (81&quot;)"
        />
        <h6
          class="MuiTypography-root makeStyles-label-5 MuiTypography-h6 MuiTypography-alignLeft"
        >
          Urban Outdoor Sofa (81")
        </h6>
        <span
          class="MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="MuiPaper-root MuiCard-root makeStyles-card-2 makeStyles-card-108 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <button
        class="MuiButtonBase-root MuiCardActionArea-root makeStyles-cardAction-3"
        tabindex="0"
        type="button"
      >
        <div
          class="MuiCardMedia-root makeStyles-media-4 makeStyles-media-109"
          style="background-image: url(https://www.westelm.com/weimgs/ab/images/wcm/products/202008/0004/urban-outdoor-lounge-chair-m.jpg);"
          title="Urban Outdoor Lounge Chair"
        />
        <h6
          class="MuiTypography-root makeStyles-label-5 MuiTypography-h6 MuiTypography-alignLeft"
        >
          Urban Outdoor Lounge Chair
        </h6>
        <span
          class="MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="MuiPaper-root MuiCard-root makeStyles-card-2 makeStyles-card-110 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <button
        class="MuiButtonBase-root MuiCardActionArea-root makeStyles-cardAction-3"
        tabindex="0"
        type="button"
      >
        <div
          class="MuiCardMedia-root makeStyles-media-4 makeStyles-media-111"
          style="background-image: url(https://www.westelm.com/weimgs/ab/images/wcm/products/202008/0019/urban-outdoor-chaise-lounger-1-m.jpg);"
          title="Urban Outdoor Chaise Lounger"
        />
        <h6
          class="MuiTypography-root makeStyles-label-5 MuiTypography-h6 MuiTypography-alignLeft"
        >
          Urban Outdoor Chaise Lounger
        </h6>
        <span
          class="MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="MuiPaper-root MuiCard-root makeStyles-card-2 makeStyles-card-112 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <button
        class="MuiButtonBase-root MuiCardActionArea-root makeStyles-cardAction-3"
        tabindex="0"
        type="button"
      >
        <div
          class="MuiCardMedia-root makeStyles-media-4 makeStyles-media-113"
          style="background-image: url(https://www.westelm.com/weimgs/ab/images/wcm/products/202005/0001/porto-outdoor-sofa-76-m.jpg);"
          title="Porto Outdoor Sofa (76&quot;)"
        />
        <h6
          class="MuiTypography-root makeStyles-label-5 MuiTypography-h6 MuiTypography-alignLeft"
        >
          Porto Outdoor Sofa (76")
        </h6>
        <span
          class="MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="MuiPaper-root MuiCard-root makeStyles-card-2 makeStyles-card-114 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <button
        class="MuiButtonBase-root MuiCardActionArea-root makeStyles-cardAction-3"
        tabindex="0"
        type="button"
      >
        <div
          class="MuiCardMedia-root makeStyles-media-4 makeStyles-media-115"
          style="background-image: url(https://www.westelm.com/weimgs/ab/images/wcm/products/202005/0003/porto-outdoor-lounge-chair-m.jpg);"
          title="Porto Outdoor Lounge Chair"
        />
        <h6
          class="MuiTypography-root makeStyles-label-5 MuiTypography-h6 MuiTypography-alignLeft"
        >
          Porto Outdoor Lounge Chair
        </h6>
        <span
          class="MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="MuiPaper-root MuiCard-root makeStyles-card-2 makeStyles-card-116 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <button
        class="MuiButtonBase-root MuiCardActionArea-root makeStyles-cardAction-3"
        tabindex="0"
        type="button"
      >
        <div
          class="MuiCardMedia-root makeStyles-media-4 makeStyles-media-117"
          style="background-image: url(https://www.westelm.com/weimgs/ab/images/wcm/products/202005/0001/coastal-outdoor-sofa-76-1-m.jpg);"
          title="Coastal Outdoor Sofa (76&quot;)"
        />
        <h6
          class="MuiTypography-root makeStyles-label-5 MuiTypography-h6 MuiTypography-alignLeft"
        >
          Coastal Outdoor Sofa (76")
        </h6>
        <span
          class="MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="MuiPaper-root MuiCard-root makeStyles-card-2 makeStyles-card-118 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <button
        class="MuiButtonBase-root MuiCardActionArea-root makeStyles-cardAction-3"
        tabindex="0"
        type="button"
      >
        <div
          class="MuiCardMedia-root makeStyles-media-4 makeStyles-media-119"
          style="background-image: url(https://www.westelm.com/weimgs/ab/images/wcm/products/202005/0001/coastal-outdoor-lounge-chair-m.jpg);"
          title="Coastal Outdoor Lounge Chair"
        />
        <h6
          class="MuiTypography-root makeStyles-label-5 MuiTypography-h6 MuiTypography-alignLeft"
        >
          Coastal Outdoor Lounge Chair
        </h6>
        <span
          class="MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="MuiPaper-root MuiCard-root makeStyles-card-2 makeStyles-card-120 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <button
        class="MuiButtonBase-root MuiCardActionArea-root makeStyles-cardAction-3"
        tabindex="0"
        type="button"
      >
        <div
          class="MuiCardMedia-root makeStyles-media-4 makeStyles-media-121"
          style="background-image: url(https://www.westelm.com/weimgs/ab/images/wcm/products/202005/0002/outdoor-slope-lounge-chair-m.jpg);"
          title="Outdoor Slope Lounge Chair"
        />
        <h6
          class="MuiTypography-root makeStyles-label-5 MuiTypography-h6 MuiTypography-alignLeft"
        >
          Outdoor Slope Lounge Chair
        </h6>
        <span
          class="MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="MuiPaper-root MuiCard-root makeStyles-card-2 makeStyles-card-122 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <button
        class="MuiButtonBase-root MuiCardActionArea-root makeStyles-cardAction-3"
        tabindex="0"
        type="button"
      >
        <div
          class="MuiCardMedia-root makeStyles-media-4 makeStyles-media-123"
          style="background-image: url(https://www.westelm.com/weimgs/ab/images/wcm/products/202006/0015/img23m.jpg);"
          title="Mexa Outdoor Lounge Chair &amp; Ottoman Set"
        />
        <h6
          class="MuiTypography-root makeStyles-label-5 MuiTypography-h6 MuiTypography-alignLeft"
        >
          Mexa Outdoor Lounge Chair & Ottoman Set
        </h6>
        <span
          class="MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
